run_unittests.py

```diff
@@ -6160,8 +6160,8 @@ def test_coverage(self):
         if not shutil.which('genhtml') and not gcovr_new_rootdir:
             raise unittest.SkipTest('genhtml not found and gcovr is too old')
         if 'clang' in os.environ.get('CC', ''):
```

Comment:
this is... utterly broken. I know it's not part of your patch, but this really needs to be:
```python
        env = get_fake_env(testdir, self.builddir, self.prefix)
        if env.detect_c_compiler(MachineChoice.HOST).get_id() == 'clang':
```